insights.go

```diff
@@ -25,11 +31,25 @@ func insightsIsRegistered() (bool, error) {
 	// The `.registered` file is always present on a registered system.
 	err := exec.Command("/usr/bin/insights-client", "--status").Run()
 	if err != nil {
+		// When the error is ExitError, then we know that insights-client only returned
+		// some error code not equal to zero. We do not care about error number.
+		var exitError *exec.ExitError
+		if errors.As(err, &exitError) {
+			// Check if .unregistered file exists. In that case insights-client
+			// is probably in non-legacy mode, and it is only unregistered.
+			// There is hopefully no reason to return any error in such case.
+			_, err := os.Stat("/etc/insights-client/.unregistered")
+			if err == nil {
+				return false, nil
+			}
+		} else {
+			return false, err
```

Comment:
We don't need this `else` branch, do we? It should ball back to the `return` just below.

Reply:
You are right.